aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1867,15 +1868,30 @@ std::tuple<Tensor, Tensor, Tensor> _lu_with_info_cuda(const Tensor& self, bool p
   Tensor pivots_tensor = at::arange(1, k + 1, self.options().dtype(at::kInt)).expand(req_size).contiguous();
   req_size.pop_back();
   auto infos_tensor = at::zeros(req_size, self.options().dtype(at::kInt));
+  auto batch_size = cuda_int_cast(batchCount(self), "batch size");
```

Comment:
If the batch_size is not passed to MAGMA/cuSOLVER/cuBLAS function it should not be cast to int.
```suggestion
  auto batch_size = batchCount(self);
```

Reply:
Revised.